# SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
# SPDX-License-Identifier: MIT
diff --git a/node_modules/@rxliuli/vista/dist/interceptors/xhr.mjs b/node_modules/@rxliuli/vista/dist/interceptors/xhr.mjs
index a1a2018..9cfb6ca 100644
--- a/node_modules/@rxliuli/vista/dist/interceptors/xhr.mjs
+++ b/node_modules/@rxliuli/vista/dist/interceptors/xhr.mjs
@@ -258,9 +258,10 @@ export const interceptXHR = function(middlewares) {
         openArgs.push(this.#password);
       }
       super.open.apply(this, openArgs);
-      Object.entries(this.#headers).forEach(([name, value]) => {
+      // https://github.com/rxliuli/vista/issues/4
+      for (const [name, value] of c.req.headers.entries()) {
         super.setRequestHeader.apply(this, [name, value]);
-      });
+      }
       this.#listeners.filter(
         ([type]) => ![
           "load",
